src/Data/Managers/V2/FormSuggestions/useSuggestions.ts

```diff
@@ -0,0 +1,80 @@
+import { useQuery } from "@tanstack/react-query";
+import { FormSuggestion } from "@/Core";
+import { PrimaryBaseUrlManager } from "@/UI";
+
+/**
+ * Custom hook to handle suggested values for a parameter.
+ * If the suggestions are literal, it will return the values directly.
+ * If the suggestions are parameters, it will fetch the parameter from the API.
+ * if the suggestions are null or undefined, it will return null as data, and a success status.
+ *
+ * @param suggestions - The suggestions for the parameter.
+ * @param environment - The environment for the parameter.
+ *
+ * @returns The result of the query, {data, status, error, isLoading}.
+ */
+export const useSuggestedValues = (
+  suggestions: FormSuggestion | null | undefined,
+  environment: string,
+) => {
+  if (!suggestions) {
+    return {
+      useOneTime: () => {
+        return { data: null, status: "success", error: null, isLoading: false };
+      },
+    };
+  }
+
+  if (suggestions.type === "literal") {
+    return {
+      useOneTime: () => {
+        return {
+          data: suggestions.values,
+          status: "success",
+          error: null,
+          isLoading: false,
+        };
+      },
+    };
+  }
+
+  const baseUrlManager = new PrimaryBaseUrlManager(
+    globalThis.location.origin,
+    globalThis.location.pathname,
+  );
+  const baseUrl = baseUrlManager.getBaseUrl(process.env.API_BASEURL);
```

Comment:
I think we could reuse urlManager here
`const { urlManager } = useContext(DependencyContext);`
`const baseUrl = urlManager.getApiUrl();`

Reply:
I wanted to avoid having the Dependency context here to keep it more simple. Adding the context would mean we need to inject it in the tests too, but if you think it is best to use it, I can change it.

Reply:
That's a fair point, I thought about having not to repeat that in every query. I don't have strong preference though, It's a trade-off and we can stand on that site, especially if we gonna remove old queries in some time in future, there is a high chance that whole `urlManager` will be redundant.

Reply:
it's two or three lines, I think for now, having it dependency-less is best, we can revisit this later. Sadly, react-query doesn't have a way to set it in the queryClient configuration, which would have been handy.